spec/tasks/seed_users_spec.rb

```diff
@@ -1,20 +1,12 @@
 require 'spec_helper'
+require_relative '../../lib/db_seed_users'
 
 RSpec.describe 'Seeding the database with users', :task do
-  #  let(:valid_attributes) { attributes_for(:user) }
-
-  before(:all) do
-    Rake.application.init
-    Rake.application.load_rakefile
-    Rake::Task.define_task(:environment)
-    Rake::Task['db:seed:users'].invoke
-  end
-
   it 'creates the users' do
+    expect { SeedUsers.run }.to output(/Inserted 1 users/).to_stdout
+
```

Comment:
This makes alot more sense than running the rake task in a before block. That code seemed like alot just to run the task.

Reply:
It is, and the line `Rake.application.load_rakefile` was running the task, too!
This is why we like abstractions around certain things. By wrapping the logic of the task in a Ruby class, you can now easily write tests for something that used to be tough to work with, and you decouple your logic from the thing that actually runs it.
You can do it, and there are some great blogs about this. Here's one I really like: https://thoughtbot.com/blog/test-rake-tasks-like-a-boss

Something like this might be good if we wanted to test a task that takes its input from a file on the command line, but for these (and this project) it's definitely overkill.

Reply:
I think this one was the base for a thing I built at Trineo, I used it a ton over the next 3 years so it was worth the effort.
https://medium.com/@jelaniwoods/rspec-tests-for-rake-tasks-da7985896014